tools/tests/replace-next-version-tag.test.sh

```
@@ -0,0 +1,131 @@
+echo "== Safety net: a token in an unsupported context is reported =="
+cat > src/unsupported.php <<'PHP'
+<?php
+// This bare string is not one of the recognized patterns.
+$note = 'shipped in $$next-version$$';
+PHP
+git add -A
+
+unsupported_exit=0
+CI='' bash "$SCRIPT" "$VERSION" . 2>/tmp/nv_stderr.$$ || unsupported_exit=$?
+if [[ "$unsupported_exit" -ne 0 ]]; then pass "exits non-zero when a token cannot be placed"; else fail "should exit non-zero for an unsupported token"; fi
+if grep -Fq 'Unexpected `$$next-version$$` token' /tmp/nv_stderr.$$; then pass "reports the offending file/line"; else fail "should report the unexpected token"; fi
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)